### PR TITLE
fix: feature resolution

### LINKED ISSUE #2271: Abigen produces bindings with dependency issues
**Version**
439a0c7de0fb8569178d56933b000d8783805c2f

**Platform**
MacOS

**Description**
https://github.com/gakonst/ethers-rs/pull/2222 seems to have affected binding generation. `forge bind` generates bindings which produce the following errors 

```
error[E0432]: unresolved import `futures_util::try_join`
   |
35 | use futures_util::{lock::Mutex, try_join};
   |                                 ^^^^^^^^ no `try_join` in the root

error: cannot determine resolution for the macro `try_join`
    |
789 |             try_join!(self.resolve_field(ens_name, "avatar"), self.resolve_name(ens_name))?;
    |             ^^^^^^^^
    |
    = note: import resolution is stuck, try simplifying macro imports

    Checking ethers-contract-abigen v2.0.0 (https://github.com/gakonst/ethers-rs#439a0c7d)
```

Relatedly, should the generated `Cargo.toml` file pin ethers to a specific version?

### LINKED ISSUE #2273: ethers no longer compiles on master
```
ethers = { git = "https://github.com/gakonst/ethers-rs", default-features = false, features = ["abigen"] }
```

```
error[E0432]: unresolved import `futures_util::try_join`
  --> /Users/Matthias/.cargo/git/checkouts/ethers-rs-c3a7c0a0ae0fe6be/18d4042/ethers-providers/src/rpc/provider.rs:35:33
   |
35 | use futures_util::{lock::Mutex, try_join};
   |                                 ^^^^^^^^ no `try_join` in the root

error: cannot determine resolution for the macro `try_join`
   --> /Users/Matthias/.cargo/git/checkouts/ethers-rs-c3a7c0a0ae0fe6be/18d4042/ethers-providers/src/rpc/provider.rs:789:13
    |
789 |             try_join!(self.resolve_field(ens_name, "avatar"), self.resolve_name(ens_name))?;
    |             ^^^^^^^^
    |
    = note: import resolution is stuck, try simplifying macro imports

For more information about this error, try `rustc --explain E0432`.
error: could not compile `ethers-providers` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0277]: `syn::Path` doesn't implement `Debug`
  --> /Users/Matthias/.cargo/git/checkouts/ethers-rs-c3a7c0a0ae0fe6be/18d4042/ethers-contract/ethers-contract-abigen/src/lib.rs:86:5
   |
64 | #[derive(Clone, Debug)]
   |                 ----- in this derive macro expansion
...
86 |     derives: Vec<syn::Path>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^ `syn::Path` cannot be formatted using `{:?}` because it doesn't implement `Debug`
   |
   = help: the trait `Debug` is not implemented for `syn::Path`
   = help: the trait `Debug` is implemented for `Vec<T, A>`
   = note: this error originates in the derive macro `Debug` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0277`.
error: could not compile `ethers-contract-abigen` (lib) due to previous error
```

@DaniPopes this looks like some weirdness with workspace deps and features

I was unable to fix this because I don't understand why this is happening at all.
I was able to get rid of the try_join error by removing the  `default-features = false` for `futures-util` in the workspace but enabling the syn features did not work change anything.
incredibly cursed.

### PR DESCRIPTION
**Pull request by DaniPopes**
## Motivation

Fixes #2271
Fixes #2273

## Solution

Explicitly specify `resolver = "2"` in the workspace
Add missing features which were getting wrongly compiled in due to resolver defaulting to "1"

## PR Checklist

-   [ ] Added Tests
-   [ ] Added Documentation
-   [ ] Updated the changelog
-   [ ] Breaking changes

### CODE REVIEW

**Comment by DaniPopes** [comment]
Can you check by setting ethers to this branch? @mattsse @FrankieIsLost

File: Cargo.toml [hunk 1]
```diff
@@ -58,6 +58,8 @@ default-members = [
     "ethers-solc",
 ]
 
+resolver = "2"
```
Comment (mattsse) [inline Cargo.toml:61]: TIL
https://doc.rust-lang.org/edition-guide/rust-2021/default-cargo-resolver.html

so if the crate does not use edition 2021 it falls back to old resolver unless the dependency uses resolver 2?

man I wonder how much time will be wasted on debugging problems this is causing

**Review by mattsse — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (DaniPopes) [reply Cargo.toml:61]: Oh yeah there it is, virtual workspaces default to v1. Crazy

**Comment by mattsse** [comment]
> Can you check by pinning `ethers` to this branch? @mattsse @FrankieIsLost
> 
> ```toml
> ethers = { git = "https://github.com/danipopes/ethers-rs", branch = "fix/feature-resolution" }
> ```

that works! ty

**Comment by FrankieIsLost** [comment]
works for me as well, thanks!

**Review by gakonst — APPROVED** [review]

**Pull request merged**